docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+    use_react_native_codegen_discovery!({
+        react_native_path: config[:reactNativePath],
+        # Modify here if your app root path isn't the same as this one.
+        app_path: "#{Dir.pwd}/..",
+        fabric_enabled: fabric_enabled,
+    })
+  end
+end
+
+// Pass the flag to enable fabric to use_react_native!.
+use_react_native!(
+  ...
+  :fabric_enabled => fabric_enabled
+)
+```
+
+```bash
+$ ~/yourproject/**/ios; pod install
+```
+
+### 2. Update your root view
+
+The way to render your app with Fabric depends on your setup. Here is an example of how you can enable Fabric in your app with the RN_FABRIC_ENABLED compiler flag to enable/disable. Refer[RN-Tester’s AppDelegate](https://github.com/facebook/react-native/blob/main/packages/rn-tester/RNTester/AppDelegate.mm) as an example.
+
+```objc
+// AppDelegate.mm
+
+#ifdef RN_FABRIC_ENABLED
+#import <React/RCTFabricSurfaceHostingProxyRootView.h>
+#import <React/RCTSurfacePresenter.h>
+#import <React/RCTSurfacePresenterBridgeAdapter.h>
+#import <react/config/ReactNativeConfig.h>
+#endif
+
+@interface AppDelegate () <RCTCxxBridgeDelegate, RCTTurboModuleManagerDelegate> {
+#ifdef RN_FABRIC_ENABLED
+  RCTSurfacePresenterBridgeAdapter *_bridgeAdapter;
+  std::shared_ptr<const facebook::react::ReactNativeConfig> _reactNativeConfig;
+  facebook::react::ContextContainer::Shared _contextContainer;
+#endif
+
+// Find a line that define rootView and replace/edit with the following lines.
+
+#ifdef RN_FABRIC_ENABLED
+  _contextContainer = std::make_shared<facebook::react::ContextContainer const>();
+  _reactNativeConfig = std::make_shared<facebook::react::EmptyReactNativeConfig const>();
+
+  _contextContainer->insert("ReactNativeConfig", _reactNativeConfig);
+
+  _bridgeAdapter = [[RCTSurfacePresenterBridgeAdapter alloc] initWithBridge:_bridge contextContainer:_contextContainer];
+
+  _bridge.surfacePresenter = _bridgeAdapter.surfacePresenter;
+
+  UIView *rootView = [[RCTFabricSurfaceHostingProxyRootView alloc] initWithBridge:_bridge
+                                                                       moduleName:@"MyTestApp"
+                                                                initialProperties:nil];
+#else
+   // Current implementation to define rootview.
+  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
+                                                   moduleName:@"MyTestApp"
+                                            initialProperties:nil];
+#endif
+```
+
+### 3. Add Babel Plugins
+
+This will trigger the codegen that will run at the metro building time.
+
+```javascript
+// babel.config.js
+module.exports = {
+  presets: ['module:metro-react-native-babel-preset'],
+  plugins: [
+    '@babel/plugin-proposal-class-properties',
+    './node_modules/react-native/packages/babel-plugin-codegen'
+  ]
+};
+```
+
+### 4. Run pod install
+
+```bash
+// Run pod install with the flags
+USE_FABRIC=1 USE_CODEGEN_DISCOVERY=1 pod install
```

Comment:
based on the nightly build by Dec 28, the playbook's ios part seems to be a bit out-of-date, e.g. I need to run ```USE_FABRIC=1 USE_CODEGEN_DISCOVERY=1 RCT_NEW_ARCH_ENABLED=1 pod install``` here to generate the codes correctly

Reply:
Yup I believe that's correct. @sota000 can you confirm?

Reply:
It should be overriding USE_CODEGEN_DISCOVERY [here](https://github.com/facebook/react-native/blob/ca5aaa766329055f206e51b2eaefcba4f282b05a/scripts/react_native_pods.rb#L31) when RCT_NEW_ARCH_ENABLED is set to 1. I'll check why it's not generate the codes correctly.

Reply:
Oh! I misread, yes, the instruction will be updated to use:
USE_FABRIC=1 RCT_NEW_ARCH_ENABLED=1 pod install

USE_CODEGEN_DISCOVERY=1 should be automatically set when passing RCT_NEW_ARCH_ENABLED=1.

Reply:
Updated 👍